mlptrain/sampling/reaction_coord.py

```diff
@@ -122,3 +140,75 @@ def _grad(self, atoms: ase.atoms.Atoms):
     def n_pairs(self) -> int:
         """Number of pairs of atoms in this set"""
         return len(self.atom_pair_list)
+
+
+class DifferenceDistance(ReactionCoordinate, _Distances):
+    """Difference in distance between two pairs of specified atoms"""
+
+    def __init__(self, *args):
+        """
+        Difference of a pair of distances e.g.
+
+        # Δr = r_1 - r_2, where r_i is the distance between atoms j and k
+        dists = DifferenceDistance((0, 1), (0, 2))
+
+        Must comprise two pairs of atoms
+
+        -----------------------------------------------------------------------
+        Arguments:
+            args: Pairs of atom indices
+        """
+
+        _Distances.__init__(self, *args)
+
+        if len(args) != 2:
+            raise ValueError('DifferenceDistance must comprise exactly two '
+                             'pairs of atoms')
+
+    def _call(self, atoms: ase.atoms.Atoms):
+        """Difference in distance between two atom pairs"""
+        dists = [atoms.get_distance(i, j, mic=True)
+                 for (i, j) in self.atom_pair_list]
+
+        return dists[0] - dists[1]
+
+    def _grad(self, atoms: ase.atoms.Atoms):
+        """Gradient of the difference in distance between two atom pairs. Each
+        component of the gradient is calculated using ∇B_i,m:
+
+        ∇B_i,m = (r_i,m - r_i,m') / ||r_mm'||
+
+        An additional term is subtracted if there are only 3 atoms in the 2
+        pairs. e.g. - (r_i,m - r_i,n) / ||r_mn||
+
+        ∇B_i,m:  Gradient of bias for atom m along component i for pair m, m'
+        r_i,m:   i (= x, y or z) position of atom in pair m, m'
+        ||r_m||: Euclidean distance between atoms in pair m, m'
+        """
+
+        derivative = np.zeros(shape=(len(atoms), 3))
+
+        distances = [atoms.get_distance(i, j, mic=True)
+                     for (i, j) in self.atom_pair_list]
+
+        for m, (i, j) in enumerate(self.atom_pair_list):
+
+            x_dist, y_dist, z_dist = [atoms[i].position[k] -
+                                      atoms[j].position[k] for k in
+                                      range(3)]
+
+            x_i = x_dist / distances[m]
+            y_i = y_dist / distances[m]
+            z_i = z_dist / distances[m]
+
+            if m == 0:
+
+                derivative[i][:] += [x_i, y_i, z_i]
+                derivative[j][:] += [-x_i, -y_i, -z_i]
+
+            elif m == 1:
+
+                derivative[i][:] += [-x_i, -y_i, -z_i]
+                derivative[j][:] += [x_i, y_i, z_i]
```

Comment:
was thinking something like:

```suggestion
    for m, (i, j) in enumerate(atom_pair_list):

        r = atoms[i].position - atoms[j].position
        r /= np.linalg.norm(r)

        sign = 1 if m == 0 else -1

        derivative[i] += sign * r
        derivative[j] += -sign * r
```